File: 18주차/NH/Partition Array Into Three Parts With Equal Sum.js
function canThreePartsEqualSum(arr) {
    let sum=arr.reduce((sum,cur)=>sum+cur)/3 ;
    let sample=0;
    let count=0;
    for(let i=0; i<arr.length; i++){
        if(count==2)return true;
        if(i==arr.length-1&&count!=2)return false;
        sample+=arr[i];
        if(sum==sample){
            sample=0;
            count++;
        }
    }
};

File: 18주차/SG/Partition_Array_Into_Three_Parts_With_Equal_Sum.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    bool canThreePartsEqualSum(vector<int>& arr) {
        int sum = 0;
        for(auto cur : arr){
            sum+=cur;
        }
        if(sum%3 !=0 ) return false;
        sum/=3;
        int subSum = 0;
        int count = 0;
        for(auto cur: arr){
            subSum+=cur;
            if(subSum  == sum){
                count++;
                subSum = 0;
            }
        }
        if(count >= 3) return true;
        return false;
    }
};